geopandas/io/tests/test_file.py

```diff
@@ -216,6 +216,34 @@ def test_to_file_schema(tmpdir, df_nybb):
     assert result_schema == schema
 
 
+def test_append_file(tmpdir, df_nybb, df_null):
+    """ Test to_file with append mode and from_file """
+    tempfilename = os.path.join(str(tmpdir), "boros.shp")
+    df_nybb.to_file(tempfilename, driver="ESRI Shapefile")
+    df_nybb.to_file(tempfilename, mode="a", driver="ESRI Shapefile")
```

Comment:
Instead of hardcoding the driver to ESRI Shapefile, how about adding the driver as a function parameter (like the rest of the tests), then skipping the test if the driver isn't in `fiona.supported_drivers`? That way you could make use of the `@pytest.mark.parametrize("driver,ext", driver_ext_pairs)`, like the rest of the tests.

It'd look something like:
```python
@pytest.mark.parametrize("driver,ext", driver_ext_pairs)
def test_append_file(tmpdir, df_nybb, df_null, driver):
    from fiona import supported_drivers
    if 'a' not in supported_drivers[driver]:
        return None
    tempfilename = os.path.join(str(tmpdir), "boros.shp")
    df_nybb.to_file(tempfilename, driver=driver)
    df_nybb.to_file(tempfilename, mode="a", driver=driver)
    # Continue with the rest of the test
```
```